frontend/src/plugin/Plugins.tsx

```diff
@@ -0,0 +1,46 @@
+import { useEffect } from 'react';
+import { useDispatch } from 'react-redux';
+import helpers from '../helpers';
+import { setPluginsLoadState } from '../redux/actions/actions';
+import { initializePlugins } from './index';
+
+/**
+ * For discovering and executing plugins.
+ */
+export default function Plugins() {
+  const dispatch = useDispatch();
+
+  // only run on first load
+  useEffect(() => {
+    const fetchData = async () => {
+      const pluginsScriptPaths = (await fetch(`${helpers.getAppUrl()}plugins/list`).then(resp =>
+        resp.json()
+      )) as string[];
+
+      const sources = await Promise.all(
+        pluginsScriptPaths.map(path =>
+          fetch(`${helpers.getAppUrl()}${path}`).then(resp => resp.text())
+        )
+      );
+
+      sources.forEach(source => {
+        (function (str: string) {
+          try {
+            const result = eval(str);
+            return result;
+          } catch (e) {
+            console.error(e);
+          }
+        }.call({}, source));
+      });
+      await initializePlugins();
```

Comment:
We're calling `store.dispatch({ type: UI_RESET_PLUGIN_VIEWS });` within this method which runs after loading the external plugins AFAICS. Won't this be a problem now that the plugins perform code right when they're loaded (before it worked on the assumption we'd load the code but only initialize the plugins after resetting views, etc.)?

Reply:
Thanks. I moved this so the initialization is done first before loading the plugins.